pkg/controller/storagecluster/reconcile.go

```diff
@@ -429,25 +430,25 @@ func (r *ReconcileStorageCluster) validateStorageDeviceSets(sc *ocsv1.StorageClu
 	return nil
 }
 
-func (r *ReconcileStorageCluster) getStorageClusterEligibleNodes(sc *ocsv1.StorageCluster, reqLogger logr.Logger) (*corev1.NodeList, error) {
-	nodes := &corev1.NodeList{}
-	selector, err := metav1.LabelSelectorAsSelector(sc.Spec.LabelSelector)
-	if err != nil {
-		return nodes, err
-	}
-	nodeMatchLabel := map[string]string{defaults.NodeAffinityKey: ""}
+func (r *ReconcileStorageCluster) getStorageClusterEligibleNodes(sc *ocsv1.StorageCluster, reqLogger logr.Logger) (nodes *corev1.NodeList, err error) {
+	nodes = &corev1.NodeList{}
+	var selector labels.Selector
+	var labelSelector *metav1.LabelSelector
+
 	if sc.Spec.LabelSelector != nil {
-		err = r.client.List(context.TODO(), nodes, MatchingLabelsSelector{Selector: selector})
-		if err != nil {
-			return nodes, err
-		}
+		labelSelector = sc.Spec.LabelSelector
 	} else {
-		err = r.client.List(context.TODO(), nodes, client.MatchingLabels(nodeMatchLabel))
-		if err != nil {
-			return nodes, err
+		labelSelector = &metav1.LabelSelector{
+			MatchLabels: map[string]string{defaults.NodeAffinityKey: ""},
 		}
 	}
-	return nodes, err
+
+	selector, err = metav1.LabelSelectorAsSelector(labelSelector)
+	err = r.client.List(context.TODO(), nodes, MatchingLabelsSelector{Selector: selector})
+	if err != nil {
+		return nodes, err
+	}
+	return nodes, nil
```

Comment:
This could be simplified to 

```suggestion
	return nodes, err
```